tests/console/yast2_vnc.pm

```diff
@@ -36,31 +36,37 @@ sub run {
 
     # check Remote Administration VNC got started
     assert_screen 'yast2_vnc_remote_administration';
-    # enable remote administration
-    send_key 'alt-a';
-    if (is_sle('<15') || is_leap('<15.0')) {
-        # open port in firewall if it is eanbaled and check network interfaces, check long text by send key right.
-        if (check_screen 'yast2_vnc_open_port_firewall', 30) {
-            send_key 'alt-p';
-            send_key 'alt-d';
-            assert_screen 'yast2_vnc_firewall_port_details';
-            send_key 'alt-e';
-            for (1 .. 5) { send_key 'right'; }
-            send_key 'alt-a';
-            send_key 'alt-o';
-        }
-        send_key 'alt-o';    # ok
+    if (check_var('ARCH', 's390x')) {
+        send_key $cmd{ok};
     }
     else {
-        assert_screen 'yast2_vnc_firewall_port_closed';
-        send_key 'alt-f';    # Open port
-        assert_screen 'yast2_vnc_firewall_port_open';
-        send_key 'alt-n';    # next
+        # enable remote administration
+        send_key 'alt-a';
+        if (is_sle('<15') || is_leap('<15.0')) {
+            # open port in firewall if it is eanbaled and check network interfaces, check long text by send key right.
+            if (check_screen 'yast2_vnc_open_port_firewall', 30) {
+                send_key 'alt-p';
+                send_key 'alt-d';
+                assert_screen 'yast2_vnc_firewall_port_details';
+                send_key 'alt-e';
+                for (1 .. 5) { send_key 'right'; }
+                send_key 'alt-a';
+                send_key 'alt-o';
+            }
+            send_key 'alt-o';    # ok
+        }
+        else {
+            assert_screen 'yast2_vnc_firewall_port_closed';
+            send_key 'alt-f';    # Open port
+            assert_screen 'yast2_vnc_firewall_port_open';
+            send_key 'alt-n';    # next
+        }
     }
     # confirm with OK for Warning dialogue
     assert_screen 'yast2_vnc_warning_text';
     send_key 'alt-o';
     wait_serial('yast2-remote-status-0', 60) || die "'yast2 remote' didn't finish";
+    return if check_var('ARCH', 's390x');
```

Comment:
Could you please add the comment here. Also, this will be the same for any remote installation, where we use VNC. @SergioAtSUSE do we have some method to check if we have VNC in the setup?

Reply:
not yet, I would have done it on this ticket: https://progress.opensuse.org/issues/38423

For sure this is a good place to start. I can reuse that function later and extend it to all Remote backends (VNC)

Reply:
PR is ready to be merged according to suggestion in https://progress.opensuse.org/issues/39332, I don't get why we are not moving forward according to the scope of this ticket, as even  this not being the most general solution for vnc, if we merge it, we unblock that other tests can continue and also with the VR we have ensured retro-compability and we are not breaking other tests.

Reply:
I'm with @jknphy, if we want to fix the world, let's plan it, schedule and execute. The only missing part is to write down this code as one of the places, where we need to remove architecture check with type of remote installation. I will talk to @SergioAtSUSE

Reply:
@okurz, I think that creating yet another settings is a bad idea. We should be able to determine if it is a **remote backend** and/or a **remote installation** based on the already existing settings.

Reply:
https://progress.opensuse.org/issues/41546 created ticket which will follow after poo#33388 to address this.

Reply:
please calm down. Sorry, I just mentioned it because I thought shouldn't be hard to extract one single method which could have been less work than creating the ticket another ticket. But if I was wrong with my assessment of course it is fine to keep the PR as is